### DISCUSSION TITLE
Logging of OIDC tokens returned from IDP

### DISCUSSION THREAD

**Discussion by gattobriz** [keycloak/keycloak#19169] [Q&A]
Hi,
is there a way in keycloak to log the content of the ID token returned from external IDP for debugging? I increased the log level to debug, but all I found was junks of JWT logged by apache and it's time consuming to manually reassemble and decode the JWT.

Thanks Gerald

**Comment by darius-m** [comment]
You can go to the client's scope page and use the evaluate tab to see the attributes that are exported, if that is what you need.

**Reply by gattobriz** [reply]
Thanks for your answer! I would need to know the content of the ID Token which is sent from the OIDC Identity Provider to  Keycloak, not the content of the tokens sent from Keycloak to the clients.
